File: output/ChatGPT/avatar/Java/Python/codeforces_567_A.py
a = int(input())
ar = list(map(int, input().split()))

for i in range(len(ar)):
    max_val = max(ar[i] - ar[0], ar[-1] - ar[i])
    if i == 0:
        min_val = ar[i + 1] - ar[i]
    elif i == len(ar) - 1:
        min_val = ar[i] - ar[i - 1]
    else:
        min_val = min(ar[i] - ar[i - 1], ar[i + 1] - ar[i])
    print(min_val, max_val)

File: output/ChatGPT/avatar/Java/C++/codeforces_567_A.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int a;
    cin >> a;
    int ar[a];
    int max = 0, min = 0;
    for (int i = 0; i < a; i++) {
        cin >> ar[i];
    }
    for (int i = 0; i < a; i++) {
        max = max(ar[i] - ar[0], ar[a - 1] - ar[i]);
        if (i == 0) {
            min = ar[i + 1] - ar[i];
        } else if (i == a - 1) {
            min = ar[i] - ar[i - 1];
        } else {
            min = min(ar[i] - ar[i - 1], ar[i + 1] - ar[i]);
        }
        cout << min << " " << max << endl;
    }
}